fsm/src/main/java/org/egov/fsm/service/EnrichmentService.java

```diff
@@ -0,0 +1,242 @@
+package org.egov.fsm.service;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.fsm.config.FSMConfiguration;
+import org.egov.fsm.repository.FSMRepository;
+import org.egov.fsm.repository.IdGenRepository;
+import org.egov.fsm.util.ComparisionUtility;
+import org.egov.fsm.util.FSMAuditUtil;
+import org.egov.fsm.util.FSMConstants;
+import org.egov.fsm.util.FSMErrorConstants;
+import org.egov.fsm.util.FSMUtil;
+import org.egov.fsm.web.model.AuditDetails;
+import org.egov.fsm.web.model.FSM;
+import org.egov.fsm.web.model.FSMAudit;
+import org.egov.fsm.web.model.FSMRequest;
+import org.egov.fsm.web.model.FSMSearchCriteria;
+import org.egov.fsm.web.model.Workflow;
+import org.egov.fsm.web.model.idgen.IdResponse;
+import org.egov.fsm.web.model.user.User;
+import org.egov.fsm.web.model.user.UserDetailResponse;
+import org.egov.tracer.model.CustomException;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Service
+@Slf4j
+public class EnrichmentService {
+	@Autowired
+	private FSMConfiguration config;
+
+	@Autowired
+	private IdGenRepository idGenRepository;
+	@Autowired
+	private BoundaryService boundaryService ;
+	
+	@Autowired
+	private FSMRepository repository ;
+
+	@Autowired
+	private UserService userService;
+	
+	@Autowired
+	private FSMUtil fsmUtil;
+	
+	@Autowired
+	private ComparisionUtility comparisionUtility;
+	/**
+	 * enrich the create FSM request with the required data
+	 * @param fsmRequest
+	 * @param mdmsData
+	 */
+	public void enrichFSMCreateRequest(FSMRequest fsmRequest, Object mdmsData) {
+		//TODO add requied logic
+		RequestInfo requestInfo = fsmRequest.getRequestInfo();
+		
+		if( fsmRequest.getRequestInfo().getUserInfo().getType().equalsIgnoreCase(FSMConstants.CITIZEN)) {
+			User citzen = new User();
+			BeanUtils.copyProperties(fsmRequest.getRequestInfo().getUserInfo(), citzen);
+			fsmRequest.getFsm().setCitizen(citzen);
+		}else {
+			userService.manageApplicant(fsmRequest);
+		}
+		
+		boundaryService.getAreaType(fsmRequest, config.getHierarchyTypeCode());
+		fsmRequest.getFsm().setStatus(FSM.StatusEnum.ACTIVE);
+		fsmRequest.getFsm().setApplicationStatus(FSMConstants.DRAFT);
+		AuditDetails auditDetails = fsmUtil.getAuditDetails(requestInfo.getUserInfo().getUuid(), true);
+		fsmRequest.getFsm().setAuditDetails(auditDetails);
+		fsmRequest.getFsm().setId(UUID.randomUUID().toString());
+		
+		fsmRequest.getFsm().setAccountId(fsmRequest.getFsm().getCitizen().getUuid());
+		
+		if (fsmRequest.getFsm().getAddress() != null) {

```

Comment:
add annotation based null check instead of checking in if else (only if its not null in all flows)